programs/cleos/main.cpp

```diff
@@ -341,7 +343,14 @@ fc::variant push_transaction( signed_transaction& trx, packed_transaction::compr
    }
 
    if (!tx_dont_broadcast) {
-      return call(push_txn_func, packed_transaction(trx, compression));
+      bool send_txn_func_supported = true;
+      if (info.server_version_string) {
+        int major, minor;
+        if (sscanf(info.server_version_string->c_str(), "v%d.%d", &major, &minor) == 2) {
```

Comment:
I'm not sure this is a good idea; not everyone who forks eosio (telos, mainnet, graymass) maintains this version string format.

Reply:
I agree with @spoonincode. Furthermore, there is a good reason not to change which RPC is called due to reasons outside of the users control (the reported version of the node cleos talks to is outside the user's control), because the different RPCs return slightly different traces.

The code should simply use `push_txn_func` when `tx_use_old_rpc` is true; otherwise it should use ``send_txn_func`.

However, it would be good to go further and recognize if the call to `send_txn_func` failed because the endpoint wasn't recognized, and instead provide the user with a more meaningful error message that recommends they use the `--use-old-rpc` option.